File: warmups/02.two-sum-2/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0, r = numbers.size() - 1;

        while (l < r) {
            int sum = numbers[l] + numbers[r];
            if (sum == target) {
                return { l+1, r+1 };
            }

            if (sum < target) {
                l++;
            } else {
                r--;
            }
        }

        // It is gauranteed that the problem would have one solution,
        // so this line will never be reached
        return { -1, -1 };
    }
};

File: warmups/02.two-sum-2/main.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let l = 0, r = numbers.length - 1;

    while (l < r) {
        let sum = numbers[l] + numbers[r];

        if (sum == target) {
            return [ l+1, r+1 ];
        }

        if (sum < target) {
            l++;
        } else {
            r--;
        }
    }

    // It is gauranteed that the problem would have one solution,
    // so this line will never be reached
    return [ -1, -1 ];
};
